RuriLib/Blocks/Interop/Methods.cs

```diff
@@ -40,14 +41,57 @@ public static string ShellCommand(BotData data, string executable, string argume
          * of writing C# code that calls these methods where necessary once it's transpiled.
          */
 
-        public static async Task<T> InvokeNode<T>(BotData data, string scriptFile, object[] parameters)
+        public static async Task<T> InvokeNode<T>(BotData data, (string, string) scriptOrFile, object[] parameters, bool isScript = false)
         {
             data.Logger.LogHeader();
-            var result = await StaticNodeJSService.InvokeFromFileAsync<T>(scriptFile, null, parameters, data.CancellationToken).ConfigureAwait(false);
+            T result;
+
+            if (isScript)
+            {
+                result = await InvokeFromStringOrCache<T>(data, scriptOrFile, parameters);
+            }
+            else
+            {
+               
+                result = await InvokeFromFile<T>(data, scriptOrFile.Item2 /*hash*/, parameters);
+            }
+
             data.Logger.Log($"Executed NodeJS script with result: {result}", LogColors.PaleChestnut);
             return result;
         }
+        private static async Task<T> InvokeFromStringOrCache<T>(BotData data, (string, string) script, object[] parameters)
+        {
+            string hash = Path.GetFileNameWithoutExtension(script.Item2).Split('/').Last();
```

Comment:
If we reached this line, `script.Item2` should contain an MD5 hash. Why is it being treated like a file path?